client/src/components/Applications/AppForm/AppFormUtils.js

```diff
@@ -117,3 +128,57 @@ export const getQueueValidation = (queue, app) => {
       }
     );
 };
+
+/**
+ * Get corrected values for a new queue
+ *
+ * Check values and if any do not work with the current queue, then fix those
+ * values.
+ *
+ * @function
+ * @param {Object} values
+ * @returns {Object} updated/fixed values
+ */
+export const getFixedValuesForUpdatedQueue = (app, values) => {
+  const fixedValues = { ...values };
+  const queue = app.exec_sys.queues.find((q) => q.name === values.batchQueue);
+  const minNode = getMinNodeCount(queue, app);
+  const maxProcessorsOnEachNode = getMaxProcessorsOnEachNode(queue);
+
+  if (values.nodeCount < minNode) {
+    fixedValues.nodeCount = minNode;
+  }
+
+  if (values.nodeCount > queue.maxNodes) {
+    fixedValues.nodeCount = queue.maxNodes;
+  }
+
+  if (
+    queue.maxProcessorsPerNode !== -1 /* e.g. Frontera rtx/rtx-dev queue */ &&
+    values.processorsOnEachNode > maxProcessorsOnEachNode
+  ) {
+    fixedValues.processorsOnEachNode = maxProcessorsOnEachNode;
+  }
+
+  /* if user has entered a time and it's somewhat reasonable (i.e. less than max time
+  for all the queues, then we should check if the time works for the new queue and update
+  it if it doesn't.
+   */
+  if (values.maxRunTime) {
+    const longestMaxRequestedTime = app.exec_sys.queues
+      .map((queue) => queue.maxRequestedTime)
+      .sort()
+      .at(-1);
+    const runtimeRegExp = new RegExp(
+      createMaxRunTimeRegex(longestMaxRequestedTime)
+    );
+    if (
+      runtimeRegExp.test(values.maxRunTime) &&
+      values.maxRunTime > queue.maxRequestedTime
```

Comment:
cool 👍 
out of curiosity, would line 177 not also cover line 176?

Reply:
176 is checking if the time entered would be valid for the queue **with the longest run time** (`runtimeRegExp` uses `longestMaxRequestedTime`) while 177 is seeing if this time exceeds the **new** queues's `maxRequestedTime`

So it is basically:
* 176:  is there a time in `values.maxRunTime`?  (but more precise, is there a somewhat legit time?)
* 177:   and is that time too large for this new queue

I ended up with this solution as it seemed to be straightforward.  It felt like the core question that needed to be answered was "did the user enter a time and do we need to fix it".  It could be empty, incorrect format or different time values ("", "foobar", "12:10:10", "48:00:00"). I considered converting the `values.MaxRunTime` to seconds and then to make a comparison.  but then would also need to validate or check that `values.MaxRunTime` had the correct form (HH:MM:SS) so thought this approach was concise and clear.  🤔 

should maybe add some comments?

Reply:
> 177: and is that time too large for this new queue

I guess I'm confused still. Wouldn't this be the only thing that matters?

Reply:
Ah, nvm I see. You're first checking that the text entered even makes sense before you compare it as a time